### DISCUSSION TITLE
Deploying ComponentDefinitions that depend on CRDs installed by the addon

### DISCUSSION THREAD

**Discussion by semmet95** [kubevela/kubevela#6152] [Q&A] [answered]
Hey everyone.
So I'm working on this addon that installs some CRDs via `kustomize` using the `apply-component` workflow step in `template.cue`. I'm also trying to install ComponentDefinitions that would be referring to the CRDs that the addon will install.

I would like to confirm if the ComponentDefinitions would require the CRDs to be installed beforehand?
In a scenario where the CRD installation workflow step takes too long to install these CRDs will the installation of ComponentDefinitions fail?

Thanks in advance for your time :)

**Comment by Somefive** [comment, marked as answer]
ComponentDefinitions do not directly depend on CRDs. You should not get any direct error. But if the ComponentDefinition renders some resources of CRD X, you will get error if the CRD X is not installed in the cluster when the ComponentDefinition is used in applications to create resources of CRD X.

In your case, you can actually parallelize the installation of CRD and ComponentDefinition. Making ComponentDefinition depends on CRD is just a soft optimization.

**Reply by semmet95** [reply]
Thanks @Somefive for the clarification. Just for confirmation's sake, unless we start deploying applications that use these component definitions, order of installation of CRDs and the ComponentDefinitions does not matter right?
